Doc/reference/datamodel.rst

```diff
@@ -436,6 +441,10 @@ Mappings
       additional examples of mapping types, as does the :mod:`collections`
       module.
 
+      .. versionchanged:: 3.7
+         Older Python versions dictionaries do not preserve insertion order.
+         This behavior was an implementation detail of CPython from 3.6.
```

Comment:
I found this note a bit confusing on first reading it. Maybe the following would be a better wording?

> Dictionaries did not preserve insertion order in versions of Python before 3.6. In CPython 3.6, insertion order was preserved, but it was considered an implementation detail at that time rather than a language guarantee.

Reply:
Agreed, it is less confusion, much more precise and concise, I've replaced the note by yours.